deepchecks/vision/segmentation_data.py

```diff
@@ -0,0 +1,193 @@
+        return TaskType.SEMANTIC_SEGMENTATION
+
+    @abstractmethod
+    def batch_to_labels(self, batch) -> List[torch.Tensor]:
+        """Extract the labels from a batch of data.
+
+        Parameters
+        ----------
+        batch : torch.Tensor
+            The batch of data.
+
+        Returns
+        -------
+        List[torch.Tensor]
+            List of torch.Tensors. Each tensor should be of same width and height of corresponding image, and its values
+            are the true class_ids of the corresponding pixels in that image.
+            Note that the tensor should 2D, as the number of channels on the original image are irrelevant to the class.
+
+        Examples
+        --------
+        >>> import numpy as np
+        ...
+        ...
+        ... def batch_to_labels(self, batch):
+        ...     # In this example, each image's label is a tensor of boolean masks, one per class_id, indicating whether
+        ...     # that pixel is of that class.
+        ...     # We would like to convert to a format where the function returns a single mask indicating the exact
+        ...     # of each pixel:
+        ...     images = batch[0]
+        ...     labels = batch[1]
+        ...     return_labels = []
+        ...
+        ...     for label, image in zip(images, labels):
+        ...         # Here, class_id "0" is "background" or "no class detected"
+        ...         ret_label = np.zeros((image.shape[0], image.shape[1]))
+        ...         # Mask to mark which pixels are already identified as classes, in case of overlap in boolean masks
+        ...         ret_label_taken_positions = np.zeros(ret_label.shape)
+        ...
+        ...         # Go over all masks of this image and transform them to a single one:
+        ...         for i in range(len(label)):
+        ...             mask = np.logical_and(np.logical_not(ret_label_taken_positions), np.array(label[i]))
+        ...             ret_label += i * mask
+        ...
+        ...             # Update the taken positions:
+        ...             ret_label_taken_positions = np.logical_or(ret_label_taken_positions, mask)
+        ...         return_labels.append(ret_label)
+        ...
+        ...     return return_labels
+        """
+        raise DeepchecksNotImplementedError('batch_to_labels() must be implemented in a subclass')
+
+    @abstractmethod
+    def infer_on_batch(self, batch, model, device) -> Sequence[torch.Tensor]:
+        """Return the predictions of the model on a batch of data.
+
+        Parameters
+        ----------
+        batch : torch.Tensor
+            The batch of data.
+        model : torch.nn.Module
+            The model to use for inference.
+        device : torch.device
+            The device to use for inference.
+
+        Returns
+        -------
+        Sequence[torch.Tensor]
+            The predictions of the model on the batch. The predictions should be in a sequence of length N containing
```

Comment:
Think we should not that the values should be proba and sum to 1

Reply:
:+1: